controllers/secret/secret_controller.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package secret
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/vmware-samples/load-balancer-operator-for-kubernetes/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"os"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetupWithManager adds this reconciler to a new controller then to the
+// provided manager.
+func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		// Watch ConfigMap resources.
+		For(&corev1.Secret{}).
+		Complete(r)
+}
+
+// SecretReconciler reconciles avi-credentials and avi-ca secrets
+type SecretReconciler struct {
+	client.Client
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	CredentialInited bool
```

Comment:
I feel like it is a bit adhoc to store the state here. A more common practice could be utilizing the `status` field in the secret.

Reply:
Its a state internal to akoo. Its used to distinguish between akoo bootup vs update on secret. Otherwise, akoo will keep shutting down itself during bootup.